battstat/battstat-upower.c

```diff
@@ -133,6 +131,7 @@ battstat_upower_cleanup( void )
   if( upc == NULL )
     return;
   
+  g_object_unref( devices );
```

Comment:
I'd re-do the code here to put `unref` calls into `if (ptr != NULL) ... unref()`
Besides, call `g_ptr_array_unref` for safety. Hell knows how GTK works.

Reply:
This is my first time touching GTK, so I'll take your word on that. :yum:

Reply:
This is my first time reviewing GTK, so best of luck.
Hell, all the code seems the same after the years.

Reply:
You must be glad that I at least have learned some C :trollface: